### PR TITLE
Fixes a few issues with Makefile (Linux)

### PR DESCRIPTION
**Pull request by JuMeBe**
- Re-assign compiling and linking options to the most suitable variables
- Confined Mac-specific environment variable setting to OSX section
- LIB_OPTION was initially set to "-shared" but then its value was overwritten, hence lost. It made linking fail.
- Changed "fPIC" to "fpic" in the course of trying to find the right set of options to solve linking problems (which turned out to be unrelated), and left it because, when it works, it tends to make smaller objects.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,7 @@ T=openssl
 
 PREFIX		?=/usr/local
 LIB_OPTION	?= -shared 
+CC			 = gcc
```
Comment (zhaozg) [inline Makefile]: please indent here, and replease = with ?=

File: Makefile [hunk 2]
```diff
@@ -22,11 +23,11 @@ SYS := $(shell gcc -dumpmachine)
 
 ifneq (, $(findstring linux, $(SYS)))
 # Do linux things
-LDFLAGS		    = -fPIC -lrt -ldl
+LDFLAGS			 = -fpic -lrt -ldl -lm
```
Comment (zhaozg) [inline Makefile]: please align indent

File: Makefile [hunk 3]
```diff
@@ -22,11 +23,11 @@ SYS := $(shell gcc -dumpmachine)
...
 OPENSSL_LIBS	?= $(shell pkg-config openssl --libs) 
 OPENSSL_CFLAGS	?= $(shell pkg-config openssl --cflags)
-CFLAGS		 = -fPIC $(OPENSSL_CFLAGS) $(LUA_CFLAGS) 
-LIB_OPTION	 = -Wl,--no-undefined
+CFLAGS			 = -fpic $(OPENSSL_CFLAGS) $(LUA_CFLAGS)
+LIB_OPTION		+= -Wl,--no-undefined
```
Comment (zhaozg) [inline Makefile]: and here

File: Makefile (hunk 1 above)
Comment (JuMeBe) [inline Makefile]: In fact, I had it "properly" indented, with spaces only, but that meant changing the previous two lines as well. The problem here is that, as soon as the width of tab changes (mine is set to 4), the alignment changes. That's why I don't like to align with tabs between parts of text (I only use tabs for left-side indentation).
So, how do you want me to indent?

Comment (zhaozg) [inline Makefile]: please keep use 4 spaces indent

Comment (JuMeBe) [inline Makefile]: If you edit the file right here, in GitHub, and set the tab size to 4, you will see that my lines are well aligned. So I can't do anything else in that sense. The problem is that you cannot always change the width of shown tabs.

**Comment by zhaozg** [comment]
Thanks @JuMeBe 's PR ,merged.

**Comment by JuMeBe** [comment]
Thanks to you, @zhaozg.

**Comment by zhaozg** [comment]
I do little change of you PR, last version is https://github.com/zhaozg/lua-openssl/blob/master/Makefile

**Pull request merged**